src/docs/release/breaking-changes/passing-generic-type-to-route-factory.md

```diff
@@ -0,0 +1,76 @@
+---
+title: Passing generic type to RouteFactory
+description: The generic type is now passed to the RouteFactory.
+---
+
+## Summary
+
+The generic type is now passed to the `RouteFactory`.
+
+## Context
+
+Previous `RouteFactory` only returns `Route<dynamic>`, typically the `onGenerateRoute` and the `onUnknownRoute` method in `MaterialApp` and `CupertinoApp`. When user called navigator and `await` for the `Navigator.pop` result with a specific type (such as boolean), but no route table defined at `routes`, the `onGenerateRoute` would try to resolve the route which the user was requesting, then it would throw a type error. With this change, `Route` will return the same route type if there's a generic type provided to `onGenerateRoute` or `onUnknownRoute`.
+
+## Description of change
+
+The generic type is added to `RouteFactory`'s `typedef`. You must pass generic type when defining `onGenerateRoute` or `onUnknownRoute`.
+
+## Migration guide
+
+You must migrate your code as described in this section
+if you're subclassing or implementing `onGenerateRoute` / `onUnknownRoute`.
+If you do, the analyzer shows the following warnings when you
+upgrade to the Flutter version that includes this change:
+
+```none
+error • The argument type 'MaterialPageRoute<dynamic> Function(RouteSettings)' can't be assigned to the parameter type 'Route<T> Function<T>(RouteSettings)' • lib/main.dart:114:7 • argument_type_not_assignable
```

Comment:
is there a cupertino version of this error too?

Reply:
Yes it does, should I put it here too?

Reply:
@AlexVincent525, not @Hixie, but if he wondered, others will, too.